sos/report/plugins/__init__.py

```diff
@@ -633,7 +633,7 @@ def timeout_from_options(self, optname, plugoptname, default_timeout):
         try:
             opt_timeout = self.get_option(optname)
             own_timeout = int(self.get_option(plugoptname))
-            if opt_timeout is None:
+            if opt_timeout is None or opt_timeout == TIMEOUT_DEFAULT:
                 _timeout = own_timeout
```

Comment:
I believe this needs to be `_timeout = default_timeout`, because here we are passing `self.plugin_timeout` as `default_timeout`, whereas `own_timeout` would be something passed to `-k plugin.timeout`.

Reply:
Either that, or this:

```python
if opt_timeout is None:
    _timeout = own_timeout
elif opt_timeout == TIMEOUT_DEFAULT:
    _timeout = default_timeout
    [...]
```

Reply:
Actually, we may want to nest the check against `TIMEOUT_DEFAULT` in the second conditional, so that we can properly override with `-k plugin.timeout` and `--plugin-timeout`:

```
diff --git a/sos/report/plugins/__init__.py b/sos/report/plugins/__init__.py
index cc5cb65b..336b4d22 100644
--- a/sos/report/plugins/__init__.py
+++ b/sos/report/plugins/__init__.py
@@ -636,7 +636,10 @@ class Plugin():
             if opt_timeout is None:
                 _timeout = own_timeout
             elif opt_timeout is not None and own_timeout == -1:
-                _timeout = int(opt_timeout)
+                if opt_timeout == TIMEOUT_DEFAULT:
+                    _timeout = default_timeout
+                else:
+                    _timeout = int(opt_timeout)
             elif opt_timeout is not None and own_timeout > -1:
                 _timeout = own_timeout
             else:
```

Reply:
Riight, I knew I havent thought all combinations.

Reply:
:+1: